cmd/wallet-web/src/router/index.js

```diff
@@ -6,15 +6,29 @@ SPDX-License-Identifier: Apache-2.0
 
 import { createRouter, createWebHistory } from 'vue-router';
 import store from '@/store';
+import { createKeyPair, gnapRequestAccess } from '@/mixins';
 import routes from './routes';
+import { computed } from 'vue/dist/vue.esm-bundler';
 
 const router = createRouter({
   history: createWebHistory(__webpack_public_path__),
   routes,
 });
 
-router.beforeEach((to, from, next) => {
+router.beforeEach(async (to, from, next) => {
   store.dispatch('agent/flushStore');
+  if (to.path === '/gnap') {
+    const gnapKeypair = await createKeyPair();
+    store.dispatch('updateGnapKeyPair', gnapKeypair);
```

Comment:
To be consistent, can we maintain the case logic?
```suggestion
    const gnapKeyPair = await createKeyPair();
    store.dispatch('updateGnapKeyPair', gnapKeyPair);
```